src/connectors/impls/gcl/writer/sink.rs

```diff
@@ -12,25 +12,222 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::time::Duration;
+
 use super::meta;
 use crate::connectors::google::AuthInterceptor;
 use crate::connectors::impls::gcl::writer::Config;
 use crate::connectors::prelude::*;
 use crate::connectors::utils::pb;
+use async_std::channel::bounded;
+use async_std::channel::Sender;
 use async_std::prelude::FutureExt;
+use async_std::task::JoinHandle;
 use googapis::google::logging::v2::log_entry::Payload;
 use googapis::google::logging::v2::logging_service_v2_client::LoggingServiceV2Client;
 use googapis::google::logging::v2::{LogEntry, WriteLogEntriesRequest};
 use gouth::Token;
 use prost_types::Timestamp;
-use std::time::Duration;
 use tonic::codegen::InterceptedService;
 use tonic::transport::{Certificate, Channel, ClientTlsConfig};
 use tonic::{Code, Status};
+use tremor_common::time::nanotime;
+
+#[allow(clippy::large_enum_variant)]
+pub(crate) enum GclEvent {
+    #[allow(dead_code)] // NOTE This is used by the tests and not intended for use in production
```

Comment:
would a `#[cfg(test)]` work here? I think i did this in some other place, but i am not 100% sure.

Reply:
Also probably good not to allow large enum variants and Box the large parts instead, `allow` is usually a sign of something being wrong :)